File: docs-examples/example-java/src/test/java/io/micronaut/jms/docs/exceptions/ErrorHandlingProducer.java
package io.micronaut.jms.docs.exceptions;

import io.micronaut.context.annotation.Requires;
import io.micronaut.jms.annotations.JMSProducer;
import io.micronaut.jms.annotations.Queue;
import io.micronaut.messaging.annotation.MessageBody;

import static io.micronaut.jms.activemq.classic.configuration.ActiveMqClassicConfiguration.CONNECTION_FACTORY_BEAN_NAME;

@Requires(property = "spec.name", value = "ErrorHandlingSpec")
@JMSProducer(CONNECTION_FACTORY_BEAN_NAME)
public interface ErrorHandlingProducer {
    @Queue("error-queue")
    void send(@MessageBody String body);
}

File: docs-examples/example-kotlin/src/test/kotlin/io/micronaut/jms/docs/exceptions/ErrorHandlingProducer.kt
package io.micronaut.jms.docs.exceptions

import io.micronaut.jms.activemq.classic.configuration.ActiveMqClassicConfiguration.CONNECTION_FACTORY_BEAN_NAME
import io.micronaut.jms.annotations.JMSProducer
import io.micronaut.jms.annotations.Queue
import io.micronaut.messaging.annotation.Body

@JMSProducer(CONNECTION_FACTORY_BEAN_NAME)
interface ErrorHandlingProducer {
    @Queue("error-queue")
    fun push(@Body message: String)
}